lib/rules/newline-before-return.js

```diff
@@ -101,10 +101,18 @@ module.exports = function(context) {
      * @private
      */
     function hasNewlineBefore(node) {
-        var tokenBefore = sourceCode.getTokenBefore(node);
-        var lineNumTokenBefore = tokenBefore.loc.end.line;
-        var lineNumNode = node.loc.start.line;
-        var commentLines = calcCommentLines(node, tokenBefore);
+        var tokenBefore = sourceCode.getTokenBefore(node),
+            lineNumNode = node.loc.start.line,
+            lineNumTokenBefore,
+            commentLines;
+
+        if (tokenBefore) {
+            lineNumTokenBefore = tokenBefore.loc.end.line;
+        } else {
+            lineNumTokenBefore = 0;     // assume "just before" start of file
```

Comment:
Nitpick (sorry, missed this on my first pass): First off, this shouldn't be a blocker for landing this. 

This comment might be a little confusing to the next person who's coming in without any context. Please let me know if you disagree, but in my mind the reason the default is `0` is two-fold: 

 - It allows us to accurately calculate the difference on line 117 (if there is no previous token, setting it to `0` makes sense because we're not subtracting any value for the nonexistent line number).
 - Since a comment can't be on line 0, `if (comment.loc.start.line === lineNumTokenBefore)` is correct.

So in my mine the `0` is not really an assumption of the location of anything, but the correct value for a nonexistent value when we do the required arithmetic for the rule.

I wonder if the comment is necessary at all? Again, very nitpicky and this shouldn't block this from being merged - just a thought. Thanks for bearing with me :)